schema/validator.go

```diff
@@ -44,6 +50,13 @@ func (v Validator) Validate(src io.Reader) error {
 		return errors.Wrap(err, "unable to read the document file")
 	}
 
+	if f := mapValidateDescendants[v]; f != nil {
```

Comment:
I think the idiomatic way to handle this is [with][1]:

    f, ok := mapValidateDescendants[v]
    if ok {
      …validate descendant…
    }

[1]: https://github.com/wking/image-tools/blob/callback-validation/validate/validate.go#L37-L44